docs/Customization/Custom-Core.rst

```diff
@@ -0,0 +1,272 @@
+as the template, but it is not recommended unless you are familiar with TileLink. 
+
+``memAXI4Node`` is an AXI4 master node and is defined as following in our example:
+
+.. literalinclude:: ../../generators/chipyard/src/main/scala/example/TutorialTile.scala
+    :language: scala
+    :start-after: DOC include start: AXI4 node
+    :end-before: DOC include end: AXI4 node
+
+where ``portName`` and ``idBits`` (number of bits to represent a port ID) are the parameter provides by the tile.
+Make sure to read :ref:`node_types` to check out what type of nodes Chipyard supports and their parameters!
+
+Also, by default, there are boundary buffers for both master and slave connections to the bus when they are leaving the tile, and you
+can override the following two functions to control how to buffer the bus requests/responses:
+
+.. code-block:: scala
+
+    protected def makeMasterBoundaryBuffers(implicit p: Parameters): TLBuffer
+    protected def makeSlaveBoundaryBuffers(implicit p: Parameters): TLBuffer
+
+You can find more information on ``TLBuffer`` in :ref:`diplomatic_widgets`.
+
+Create Implementation Class
+---------------------------
+
+The implementation class contains the parameterized, actual hardware that depends on the values resolved by the Diplomacy
+framework according to the info provided in the Tile class. This class will normally contains Chisel RTL code. If your
+core is in Verilog, you will need to instantiate the black box class that wraps your Verilog implementation and connect it with the buses
+and other components. No Diplomacy/TileLink code should be in this class; you should only connect the IO signals in TileLink
+interfaces or other diplomatically defined components, which are located in the tile class. 
+
+The implementation class for your core is of the following form:
+
+.. literalinclude:: ../../generators/chipyard/src/main/scala/example/TutorialTile.scala
+    :language: scala
+    :start-after: DOC include start: Implementation class
+    :end-before: DOC include end: Implementation class
+
+If you create an AXI4 node (or equivalents), you will need to connect them to your core. You can connect a port like this:
+
+.. literalinclude:: ../../generators/chipyard/src/main/scala/example/TutorialTile.scala
+    :language: scala
+    :start-after: DOC include start: AXI4 connect
+    :end-before: DOC include end: AXI4 connect
+
+Connect Interrupt
+-----------------
+
+Chipyard allows a tile to either receive interrupts from other devices or initiate interrupts to notify other cores/devices. 
+In the tile that inherited ``SinksExternalInterrupts``, one can create a ``TileInterrupts`` object (a Chisel bundle) and 
+call ``decodeCoreInterrupts`` with the object as the argument. Note that you should call this function in the implementation
+class since it returns a Chisel bundle used by RTL code. You can then read the interrupt bits from the resulting object.
```

Comment:
What is the resulting object? is that not the original TileInterrupts bundle you created?
Can you give an example code sequence, assuming I had some chisel blackbox with an `int_in` port? (this should be 3 lines)

Reply:
It's actually the ``TileInterrupts`` created in last step, and I clarified this by pointing this out in the docs. I also added some code to demonstrate how to connect interrupts and raise interrupts.